pkg/proxy/topology_test.go

```diff
@@ -200,6 +202,19 @@ func Test_filterEndpointsWithHints(t *testing.T) {
 			{ip: "10.1.2.3", zoneHints: sets.NewString("zone-a")},
 			{ip: "10.1.2.6", zoneHints: sets.NewString("zone-a")},
 		},
+	}, {
+		name:            "unready endpoint",
```

Comment:
can you add a test case for when the zone has _only_ unready endpoints?

Reply:
Good idea, added.